### ISSUE TITLE
Add Event Handlers for when we send a message to the server

### ISSUE THREAD

**Issue by xforever1313** [xforever1313/Chaskis#57]
We have event handlers that implement from IIrcHandler that are used when we receive messages from the server.  However, we should also add events when we send things to the server.  This way, we can keep track of the number of messages we send.  Perhaps we want to record this in a log or something?

We should have a handler when we send a:
* Join
* Part
* Quit
* Ping
* Pong
* Ctcp Ping
* Ctcp Pong
* Notice
* Action
* Privmsg
* All / Raw

Basically, all of our Send() functions on IrcConnection.  These events should be put on the StringParsingQueue, and include in them the server we wrote to and the protocol.

**Comment by xforever1313** [comment]
All desired events have been added to Chaskis.Core.  Closing!
